src/VisualStudio/Core/Def/Implementation/PullMemberUp/DependentsBuilder.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.  
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.PullMemberUp
+{
+    internal class DependentsBuilder : OperationWalker
+    {
+        private readonly HashSet<ISymbol> _dependents;
+
+        private readonly ImmutableHashSet<ISymbol> _membersInType;
+
+        internal static IEnumerable<ISymbol> Build(
+            SemanticModel semanticModel,
+            ISymbol symbol,
+            ImmutableHashSet<ISymbol> membersInType)
+        {
+            var builder = new DependentsBuilder(membersInType);
+
+            var operatons = symbol.DeclaringSyntaxReferences.Select(@ref => semanticModel.GetOperation(@ref.GetSyntax()));
```

Comment:
this won't work if your symbol is partial and has parts in other files.  Add test for that.